pkg/pgmodel/end_to_end_tests/promql_endpoint_integration_test.go

```diff
@@ -673,3 +57,63 @@ func testRequest(req *http.Request, handler http.Handler, client *http.Client, c
 		}
 	}
 }
+
+func testRequestConcurrent(reqs []*http.Request, logs []string, handler http.Handler, client *http.Client, comparator resultComparator) func(*testing.T) {
```

Comment:
I would consider changing reqs `[]*http.Request, logs []string` to a `cases []*testCase` where the testCase struct contains one http.Request and a log string.